File: app/src/main/java/com/kunzisoft/keepass/database/element/PwNodeInterface.java
package com.kunzisoft.keepass.database.element;

import android.os.Parcelable;

import com.kunzisoft.keepass.database.SmallTimeInterface;

public interface PwNodeInterface extends SmallTimeInterface, Parcelable {

	PwNodeId getNodeId();

	void setNodeId(PwNodeId id);

	String getTitle();

	void setTitle(String title);

	/**
	 * @return Visual icon
	 */
	PwIcon getIcon();

	void setIcon(PwIcon icon);

	/**
	 * @return Type of Node
	 */
	PwNodeInterface.Type getType();

	/**
	 * Retrieve the parent node
	 * @return PwGroup parent as group
	 */
	PwGroupInterface getParent();

	/**
	 * Assign a parent to this node
	 */
	void setParent(PwGroupInterface prt);

	boolean containsParent();

	void touch(boolean modified, boolean touchParents);

	boolean isContainedIn(PwGroupInterface container);

	boolean isSearchingEnabled();

    boolean containsCustomData();

	/**
	 * Type of available Nodes
	 */
	enum Type {
		GROUP, ENTRY
	}
}

File: app/src/main/java/com/kunzisoft/keepass/database/element/PwNodeInterface.kt
package com.kunzisoft.keepass.database.element

import android.os.Parcelable

interface PwNodeInterface<ParentGroup> : NodeTimeInterface, Parcelable {

    var title: String

    /**
     * @return Visual icon
     */
    var icon: PwIcon

    /**
     * @return Type of Node
     */
    val type: Type

    /**
     * Retrieve the parent node
     */
    var parent: ParentGroup?

    val isSearchingEnabled: Boolean

    fun containsParent(): Boolean

    fun afterAssignNewParent()

    fun isContainedIn(container: ParentGroup): Boolean

    fun touch(modified: Boolean, touchParents: Boolean)
}